src/lamp_py/performance_manager/flat_file.py

```diff
@@ -114,33 +119,83 @@ def write_flat_files(db_manager: DatabaseManager) -> None:
-    )
-    process_logger.log_start()
+        # check the file version, deleting records if they need to be replaced
+        check_version()
 
-    for service_date in service_dates:
-        sub_process_logger = ProcessLogger(
-            "flat_file_write", service_date=service_date.strftime("%Y-%m-%d")
-        )
-        sub_process_logger.log_start()
+        # get the service dates that need to be archived
+        service_dates = dates_to_update(db_manager)
 
-        try:
-            write_daily_table(db_manager=db_manager, service_date=service_date)
-        except Exception as e:
-            sub_process_logger.log_failure(e)
-        else:
-            sub_process_logger.log_complete()
+        process_logger.add_metadata(date_count=len(service_dates))
 
-    write_csv_index()
+        # if no data to archive, early exit
+        if len(service_dates) == 0:
+            process_logger.log_complete()
+            return
 
-    process_logger.log_complete()
+        for service_date in service_dates:
+            sub_process_logger = ProcessLogger(
+                "flat_file_write",
+                service_date=service_date.strftime("%Y-%m-%d"),
+            )
+            sub_process_logger.log_start()
+
+            try:
+                write_daily_table(
+                    db_manager=db_manager, service_date=service_date
+                )
+            except Exception as e:
+                sub_process_logger.log_failure(e)
+            else:
+                sub_process_logger.log_complete()
+
+        write_csv_index()
+
+        process_logger.log_complete()
+
+    except Exception as e:
+        process_logger.log_failure(e)
+
+
+def check_version() -> None:
+    """
+    check the version of of the index csv file. if it is behind the current
+    version, delete all of the files with the rail performance manager prefix.
+    """
+    index_object = os.path.join(
+        S3Archive.BUCKET_NAME,
+        S3Archive.RAIL_PERFORMANCE_PREFIX,
+        S3Archive.INDEX_FILENAME,
+    )
+
+    # get the version of the index from the metadata. a 404 will happen if the
+    # file doesn't exist, set the version to None in that case.
+    try:
+        version = object_metadata(index_object).get(S3Archive.VERSION_KEY)
+    except ClientError as error:
+        if error.response["Error"]["Code"] == "404":
+            version = None
+        else:
+            raise
+
+    # if the versions mismatch, delete everything with the rail performance
+    # manager flat file prefix.
+    if version != S3Archive.RPM_VERSION:
+        files_to_remove = file_list_from_s3(
+            bucket_name=S3Archive.BUCKET_NAME,
+            file_prefix=S3Archive.RAIL_PERFORMANCE_PREFIX,
+        )
+
+        for file in files_to_remove:
+            success = delete_object(file)
+            if not success:
+                raise RuntimeError(
+                    f"Failed to delete {file} when updating flat files"
+                )
```

Comment:
new function that checks the metadata of the csv file for the version. if it detects a version mismatch it deletes everything in the "directory", which will cause all archive files to be re-written.